sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/StaleResourceRetryPolicy.java

```diff
@@ -120,6 +127,29 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
                         }
 
                         return Mono.just(ShouldRetryResult.retryAfter(Duration.ZERO));
+                    })
+                    .onErrorMap(throwable -> {
+
+                        if (throwable instanceof CosmosException) {
```

Comment:
hmm, not sure this is the place for status code remapping since this is a retry policy - feels this logic should belong to RxCollectionCache only

Reply:
Pushed logic to RxCollectionCache.